lib/src/network/coap_inetwork.dart

```diff
@@ -50,14 +50,40 @@ abstract class CoapINetwork {
     required CoapInternetAddress? address,
     required DefaultCoapConfig config,
     String namespace = '',
+    PskCredentials? pskCredentials,
+    EcdsaKeys? ecdsaKeys,
   }) {
     int? port = uri.port > 0 ? uri.port : null;
     switch (uri.scheme) {
       case CoapConstants.uriScheme:
         return CoapNetworkUDP(address, port ?? config.defaultPort,
             namespace: namespace);
+      case CoapConstants.secureUriScheme:
+        return _determineDtlsBackend(address, port, config,
+            namespace: namespace,
+            pskCredentials: pskCredentials,
+            ecdsaKeys: ecdsaKeys);
       default:
         throw UnsupportedProtocolException(uri.scheme);
     }
   }
+
+  /// Determines which [DtlsBackEnd] to use for secure communication.
+  static CoapINetwork _determineDtlsBackend(
```

Comment:
Should we support multiple backends or should we pick the one we think is best and stick with that? Would make the code a bit simpler, or we are worried about platform support? It's a shame there are only bindings in Dart for DTLS from what I could see, nothing native :(

Reply:
Tinydtls does not support certificates, while OpenSSL does not support Raw Public Keys, yet :/ The OpenSSL binding by @Sorunome et al. is currently also missing support for Pre-Shared Keys (I tried to [integrate the feature](https://gitlab.com/famedly/company/frontend/libraries/dtls/-/merge_requests/2) in the upstream, but the project seems dormant right now). So both have features the other one is missing, which is why I thought about having multiple backends in the first place.

What I had in my mind was having both the tinydtls backend and the OpenSSL one for a certain period of time in parallel in order to cover all CoAPS features and then gradually replace them once native DTLS support becomes available. I actually also started experimenting a bit with integrating DTLS into the Dart SDK itself (which would most definitely be the cleanest solution), but figured that it might take some more time to get it into the upstream. The integration of tinydtls would therefore be a bit of a temporary workaround that hopefully can be replaced in the not too distant future.

Reply:
I now removed the backend logic and let the library use tinydtls if a PSK or ECDSA keys are provided. This way, users don't need to bother with the internals and the API can probably be kept stable, even if the backend is swapped out or another one for certificates is added.

Reply:
Maybe the method's name could now be changed to `_determineDtlsNetwork`?

Reply:
I renamed the method now to `_determineDtlsNetwork´ :)